app/lib/proxy_api/ansible.rb

```diff
@@ -53,5 +53,61 @@ def playbooks(playbooks_names = [])
     rescue *PROXY_ERRORS => e
       raise ProxyException.new(url, e, N_('Unable to get playbooks from Ansible'))
     end
+
+    def repo_information(vcs_url)
+      parse(get("vcs_clone/repo_information?vcs_url=#{vcs_url}"))
+    rescue *PROXY_ERRORS, RestClient::Exception => e
+      raise e unless e.is_a? RestClient::RequestFailed
+      case e.http_code
+      when 400
+        raise Foreman::Exception.new N_('Error requesting repo-info. Check Smartproxy log.')
+      else
+        raise e
```

Comment:
I think this causes a different trace back and you can simply use it without arguments to raise it again:
```suggestion
        raise
```